src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -125,7 +125,11 @@ public State(Vertex vertex, Edge backEdge, long timeSeconds, long startTime, Rou
             this.stateData.bikeParked = options.arriveBy;
             this.stateData.nonTransitMode = this.stateData.bikeParked ? TraverseMode.WALK
                     : TraverseMode.BICYCLE;
+        } else if (options.rideAndKiss) {
+            this.stateData.carParked = !options.arriveBy;
```

Comment:
I get that this is being done this way so that it works well with the InterleavedBidirectionalHeuristic, but it is not intuitive at first glance.  If someone were to read this they would say that when departing on transit the car is not parked in the initial state.  If this also applied to "Ride and Park" queries, it certainly wouldn't make sense as there is an expectation that the car should already be parked at the park and ride.  But that isn't really what is happening here, it's more like has a "kiss" or transition to car happened yet in the search.  At the very least, some comments ought to be added to explain the meaning of what `carParked` is keeping track of with "ride and park/kiss" enabled in "depart at" queries.  Maybe even change the `carParked` variable to something that better reflects what is being tracked.